src/test/scala/io/iohk/ethereum/jsonrpc/server/http/JsonRpcHttpServerSpec.scala

```diff
@@ -92,13 +100,119 @@ class JsonRpcHttpServerSpec extends AnyFlatSpec with Matchers with ScalatestRout
+    val postRequest =
+      HttpRequest(HttpMethods.POST, uri = "/", entity = HttpEntity(MediaTypes.`application/json`, jsonRequest))
+
+    postRequest ~> Route.seal(mockJsonRpcHttpServerWithIpRestriction.route) ~> check {
+      status shouldEqual StatusCodes.OK
+      responseAs[String] shouldEqual """{"jsonrpc":"2.0","result":"this is a response","id":1}"""
+    }
+  }
+
+  it should "return too many requests error with ip-restriction enabled and two requests executed in a row" in new TestSetup {
+    (mockJsonRpcController.handleRequest _)
+      .expects(*)
+      .returning(Task.now(JsonRpcResponse("2.0", Some(JString("this is a response")), None, JInt(1))))
+
+    val jsonRequest = ByteString("""{"jsonrpc":"2.0", "method": "asd", "id": "1"}""")
+    val postRequest =
+      HttpRequest(HttpMethods.POST, uri = "/", entity = HttpEntity(MediaTypes.`application/json`, jsonRequest))
+
+    postRequest ~> Route.seal(mockJsonRpcHttpServerWithIpRestriction.route) ~> check {
+      status shouldEqual StatusCodes.OK
+      responseAs[String] shouldEqual """{"jsonrpc":"2.0","result":"this is a response","id":1}"""
+    }
+    postRequest ~> Route.seal(mockJsonRpcHttpServerWithIpRestriction.route) ~> check {
+      status shouldEqual StatusCodes.TooManyRequests
+    }
+  }
+
+  it should "return method not allowed error for batch request with ip-restriction enabled" in new TestSetup {
+    (mockJsonRpcController.handleRequest _)
+      .expects(*)
+      .twice()
+      .returning(Task.now(JsonRpcResponse("2.0", Some(JString("this is a response")), None, JInt(1))))
+
+    val jsonRequest =
+      ByteString("""[{"jsonrpc":"2.0", "method": "asd", "id": "1"}, {"jsonrpc":"2.0", "method": "asd", "id": "2"}]""")
+    val postRequest =
+      HttpRequest(HttpMethods.POST, uri = "/", entity = HttpEntity(MediaTypes.`application/json`, jsonRequest))
+
+    postRequest ~> Route.seal(mockJsonRpcHttpServerWithIpRestriction.route) ~> check {
+      status === StatusCodes.MethodNotAllowed
+    }
+  }
+
+  it should "accept json request after rejected request with ip-restriction enabled once time has passed" in new TestSetup {
+    (mockJsonRpcController.handleRequest _)
+      .expects(*)
+      .twice()
+      .returning(Task.now(JsonRpcResponse("2.0", Some(JString("this is a response")), None, JInt(1))))
+
+    val jsonRequest = ByteString("""{"jsonrpc":"2.0", "method": "asd", "id": "1"}""")
+    val postRequest =
+      HttpRequest(HttpMethods.POST, uri = "/", entity = HttpEntity(MediaTypes.`application/json`, jsonRequest))
+
+    postRequest ~> Route.seal(mockJsonRpcHttpServerWithIpRestriction.route) ~> check {
+      status shouldEqual StatusCodes.OK
+      responseAs[String] shouldEqual """{"jsonrpc":"2.0","result":"this is a response","id":1}"""
+    }
+    postRequest ~> Route.seal(mockJsonRpcHttpServerWithIpRestriction.route) ~> check {
+      status shouldEqual StatusCodes.TooManyRequests
+    }
+
+    FakeClock.advanceTime(10)
```

Comment:
Minor: maybe we should set this to `2* config.minRequestInterval` instead of the hardcoded 10?